### PR TITLE
[SPARK-47562][CONNECT] Factor literal handling out of `plan.py`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Factor literal handling out of `plan.py`

### Why are the changes needed?

we should not do the parameter pre-processing in `plan.py`

### Does this PR introduce _any_ user-facing change?
no, just refactoring

### How was this patch tested?
ci

### Was this patch authored or co-authored using generative AI tooling?
no

### CODE REVIEW

File: python/pyspark/sql/connect/plan.py [hunk 1]
```diff
@@ -1120,53 +1126,48 @@ def plan(self, session: "SparkConnectClient") -> proto.Relation:
 
 
 class SQL(LogicalPlan):
-    def __init__(self, query: str, args: Optional[Union[Dict[str, Any], List]] = None) -> None:
+    def __init__(
+        self,
+        query: str,
+        args: Optional[List[Column]] = None,
```
Comment (xinrong-meng) [inline python/pyspark/sql/connect/plan.py:1132]: Irrevevant to changes proposed in this PR, I noticed a mix use of `List` and `Sequence` type hints in the file, but not necessarily indicating the mutability of variable.

**Comment by xinrong-meng** [comment]
LGTM, thank you!

**Review by xinrong-meng — APPROVED** [review]

File: python/pyspark/sql/connect/dataframe.py [hunk 2]
```diff
@@ -1443,8 +1443,25 @@ def all_of_(xs: Iterable) -> bool:
                 message_parameters={},
             )
 
+        def _convert_int_to_float(v: Any) -> Any:
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/dataframe.py:1446]: no need to do `_` prefix. it's nested func so not accessible

**Review by HyukjinKwon — APPROVED** [review]

File: python/pyspark/sql/connect/plan.py (hunk 1 above)
Comment (xinrong-meng) [reply python/pyspark/sql/connect/plan.py:1132]: That's good to know, sounds good!

**Comment by zhengruifeng** [comment]
thanks, merged to master

**Pull request closed without merging**